purchase_request_procurement/models/procurement.py

```diff
@@ -69,3 +70,28 @@ def _run(self, procurement):
             procurement.message_post(body=_("Purchase Request extended."))
             return True
         return super(Procurement, self)._run(procurement)
+
+    @api.model
+    def propagate_cancel(self, procurement):
+        result = super(Procurement, self).propagate_cancel(procurement)
+        # Remove the reference to the request_id from the procurement order
+        request = procurement.request_id
+        procurement.write({'request_id': None})
+        # Search for purchase request lines containing the procurement_id
+        request_lines = self.env['purchase.request.line'].\
+            search([('procurement_id', '=', procurement.id)])
+        # Remove the purchase request lines, if the request is not draft
+        # or reject
+        for line in request_lines:
+            if line.request_id.state not in ('draft', 'reject'):
```

Comment:
This "if" restriction could also be used into a filtered on request_lines in order to loop only into the wanted records